src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -408,6 +412,34 @@ protected override void StopProcessing()
 
         #region Helper Methods
 
+        private void ThrowNotSuccessStatusCodeError(HttpRequestMessage request, HttpResponseMessage response)
+        {
+            var detailMsg = "";
+            var reader = new StreamReader(StreamHelper.GetResponseStream(response));
+            try
+            {
+                detailMsg = reader.ReadToEnd();
+            }
+            catch (IOException)
+            {
+            }
+            catch (ArgumentException)
+            {
+            }
+            finally
+            {
+                reader.Dispose();
+            }
+
+            var msg = string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.NotSuccessStatusCode, (int) response.StatusCode, response.ReasonPhrase);
+            ErrorRecord er = new ErrorRecord(new WebException(msg) { Response = response }, "WebCmdletWebResponseException", ErrorCategory.InvalidOperation, request);
```

Comment:
I'm thinking aloud here: how about we add the response object to "Exception.Data" for both full and core PS? Users can get the status code by `$e.Exception.Data.Response`.
Yes, this is not backward compatible, but we are sort of not backward compatible anyways, for example, members/methods exposed by `(invoke-webrequest www.bing.com).BaseResponse` are different.